src/m365/teams/commands/message/message-restore.spec.ts

```diff
@@ -0,0 +1,229 @@
+    const actual = await command.validate({
+      options: {
+        id: messageId,
+        channelId: channelId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if teamId and teamName options are both passed', async () => {
+    const actual = await command.validate({
+      options: {
+        id: messageId,
+        teamId: teamId,
+        teamName: teamName,
+        channelId: channelId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if channelId or channelName options are not passed', async () => {
+    const actual = await command.validate({
+      options: {
+        id: messageId,
+        teamId: teamId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if channelId and channelName options are both passed', async () => {
+    const actual = await command.validate({
+      options: {
+        id: messageId,
+        teamId: teamId,
+        channelName: channelName,
+        channelId: channelId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the teamId is not a valid guid', async () => {
+    const actual = await command.validate({
+      options: {
+        teamId: "5f5d7b71-1161-44",
+        channelId: channelId,
+        id: messageId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('validates for a correct input', async () => {
+    const actual = await command.validate({
+      options: {
+        teamId: teamId,
+        channelId: channelId,
+        id: messageId
+      }
+    }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation for an incorrect channelId missing leading 19:.', async () => {
+    const actual = await command.validate({
+      options: {
+        teamId: teamId,
+        channelId: '552b7125655c46d5b5b86db02ee7bfdf@thread.skype',
+        id: messageId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation for an incorrect channelId missing trailing @thread.skype.', async () => {
+    const actual = await command.validate({
+      options: {
+        teamId: teamId,
+        channelId: '19:552b7125655c46d5b5b86db02ee7bfdf@thread',
+        id: messageId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('restores the specified message', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/teams/${teamId}/channels/${channelId}/messages/${messageId}/undoSoftDelete`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        teamId: teamId,
+        channelId: channelId,
+        id: messageId
+      }
+    });
```

Comment:
This test is missing an assert. A test without assert is actually not testing anything.
Let's check if the post request was actually fired.